### DISCUSSION TITLE
Check if store is already initialized, if not fetch data

### DISCUSSION THREAD

**Discussion by alexander-densley** [pmndrs/zustand#2340] [Q&A]
Im working on trying to implement zustand into my product, and one of the things I want to do is only fetch data from the database if it hasn't already been fetched, is there a best practice here?

**Comment by dbritto-dev** [comment]
@alexander-densley you should use react-query

**Reply by alexander-densley** [reply]
So theres no simple check for the zustand store?

**Reply by alexander-densley** [reply]
I don't need a ton of server async state management, i primarily need client side stuff, but I just want to only fetch one time

**Reply by alexander-densley** [reply]
would it be bad practice to have a variable in my store like (isFetched) that i check?

**Reply by dbritto-dev** [reply]
@alexander-densley hmmm I wouldn't say that's simple, first you need to check if there's no data then request the data then you need to handle any errors/exceptions. In a nutshell you should manage the state transitions from -> idle, fetching, success or error

**Reply by dbritto-dev** [reply]
but if you want something simple you could use `useEffect` and `setState` or `set`

**Reply by dbritto-dev** [reply]
https://www.youtube.com/watch?v=DocXo3gqGdI

**Reply by alexander-densley** [reply]
Are you recommending react query would fit my use case better over zustand? Or that I should use them together

**Reply by dbritto-dev** [reply]
server states -> react query, app/component states -> zustand

**Comment by dai-shi** [comment]
> Check if store is already initialized, if not fetch data

For example, what's wrong with this?
```js
const useDataStore = create((set) => ({
  data: null,
  fetchDataIfNull: async () => {
    if (data === null) {
      const res = await fetch(/* ... */)
      const data = await res.json()
      set({ data })
    }
  },
}))
useDataStore.getState().fetchDataIfNull()
```
